gateway/gateway.go

```diff
@@ -340,30 +446,16 @@ func (gateway *HandleT) isWriteKeyEnabled(writeKey string) bool {
 	return true
 }
 
-//Function to batch incoming web requests
-func (gateway *HandleT) webRequestBatcher() {
-	var reqBuffer = make([]*webRequestT, 0)
-	timeout := time.After(batchTimeout)
-	for {
-		select {
-		case req := <-gateway.webRequestQ:
-			//Append to request buffer
-			reqBuffer = append(reqBuffer, req)
-			if len(reqBuffer) == maxBatchSize {
-				breq := batchWebRequestT{batchRequest: reqBuffer}
-				gateway.batchRequestQ <- &breq
-				reqBuffer = nil
-				reqBuffer = make([]*webRequestT, 0)
-			}
-		case <-timeout:
-			timeout = time.After(batchTimeout)
-			if len(reqBuffer) > 0 {
-				breq := batchWebRequestT{batchRequest: reqBuffer}
-				gateway.batchRequestQ <- &breq
-				reqBuffer = nil
-				reqBuffer = make([]*webRequestT, 0)
-			}
+//Function to route incoming web requests to userWebRequestBatcher
+func (gateway *HandleT) webRequestRouter() {
+	for req := range gateway.webRequestQ {
+		userIDHeader := req.request.Header.Get("anonymousId")
+		//If necessary fetch userID from request body.
+		if userIDHeader == "" {
+			userIDHeader = uuid.NewV4().String()
```

Comment:
We should add the backup logic if the header is not present before merging to master.